WinPort/src/Backend/WX/wxWinTranslations.cpp

```diff
@@ -421,47 +421,61 @@ static int X11KeyCodeLookupUncached(wxUint32 keyflags)
 {
 	int key_code = 0;
 	Display *display = XOpenDisplay(NULL);
-	XkbDescPtr xkb = XkbGetMap(display, 0, XkbUseCoreKbd);
+
+	if (!display) {
+		return 0;	
+	}
+
+	char keycodes[] = "evdev";
+	char types[] = "complete";
+	char compat[] = "complete";
+	char symbols[] = "pc+us+inet(evdev)";
+
+	XkbComponentNamesRec component_names = {
+		.keycodes = keycodes,
+		.types = types,
+		.compat = compat,
+		.symbols = symbols
+	};
+
+	XkbDescPtr xkb = XkbGetKeyboardByName(
+		display, XkbUseCoreKbd, &component_names, XkbGBN_AllComponentsMask, XkbGBN_AllComponentsMask, False);
+
+	if (!xkb) {
+		XCloseDisplay(display);
+		return 0;	
+	}
+
 	XkbGetControls(display, XkbGroupsWrapMask, xkb);
 	XkbGetNames(display, XkbGroupNamesMask, xkb);
 
-	// searching for group id of English keyboard layout
-	int group = -1;
-	for (int i = 0; i < xkb->ctrls->num_groups; i++) {
-		char *layout = XGetAtomName(display, xkb->names->groups[i]);
-		if (strstr(layout, "English")) {
-			// English kb layout found, let's use it for translations
-			group = i;
-			break;
-		}
-		XFree(layout);
-	}
+	KeySym ks;
+	unsigned int mods;
+	XkbTranslateKeyCode(xkb, keyflags, 0, &mods, &ks);
```

Comment:
missing error check: XkbTranslateKeyCode returns True if the translation resulted in a keysym, and False if it resulted in NoSymbol.

Reply:
oh, thanks for noticing and fixing!